libhdf5/hdf5set_format_compatibility.c

```diff
@@ -0,0 +1,74 @@
+/* Copyright 2022, University Corporation for Atmospheric Research.
+ * See COPYRIGHT file for copying and redistribution conditions. */
+/**
+ * @file
+ * @internal This function selects the best HDF5 file format options
+ * to create netCDF-4 files that can be read and written by older
+ * library versions.
+ *
+ * Format compatibility is transient, not baked in to an HDF5 file
+ * at creation time.  Therefore the desired compatibilty options
+ * must be selected every time a file is opened for writing.
+ *
+ * This function should be called before every call to create a new
+ * netCDF-4 file, or to open an existing netCDF-4 file for writing.
+ * This function has no effect when opening a file for read only.
+ *
+ * This function should work correctly with all HDF5 library versions
+ * from 1.8.0 through 1.13.0 and beyond, with no further changes.
+ * This assumes that HDF5 versioning controls remain consistent
+ * into the future.
+ *
+ * The basic functionality is to select the traditional HDF5 v1.8
+ * format compatibility, whenever possible.  The less desirable
+ * v1.6 compatibily is selected in a few strange cases when it is
+ * not possible to select v1.8.
+ *
+ * Files created with v1.6 compatibility have superblock version 0.
+ * Files created with v1.8 compatibility have superblock version 2.
+ *
+ * The superblock version is locked in when a file is first created.
+ * It is then possible to get a mix of v1.6 and v1.8 internal
+ * object versions, when an existing netCDF-4 file is modified by
+ * a different software version than the one that originally
+ * created the file.  Mixed-object files of this nature are common
+ * and do not suffer any serious problems.
+ *
+ * See netcdf-c github issues #250 and #951 for more details about
+ * the rationale and evolution of netCDF-4 format compatibility.
+ */
```

Comment:
Thanks for the excellent documentation. Please add an author tag at the file and the function level, with your name, and a date tag would also be good.

Reply:
Will do.  There are potentially two authors.  I am novice with Doxygen, surviving by pure imitation, so far.  Would you please point me to appropriate documentation, if available?  The Doxygen website does not match the tag style that I have seen within netcdf-c.

Reply:
Never mind, I figured out authors.  Pure imitation saves the day.  What does a date tag look like?

Reply:
For example just add something like this at the bottom of the doxygen block...

` * @author Stephen Gilbeert @date 2002-10-28
`